server/bootstrap/passport/strategies/bnet.js

```diff
@@ -10,29 +11,27 @@ module.exports = function(authBnetConfig){
         user
           .findByUserId(profile.id)
           .then(function(){
-            user.setValues({
-                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
-                oauthType: "bnet"
-            }).updateTimeLatestLogin();
-            if (!user.exists()) {
-              user.setValues({
-                  userId: profile.id
-              });
-            }
-            user
-              .save()
-              .then(function(){
-                console.log('Save: success', user.getValues());
-                return done(null, profile);
-              })
-              .error(function(err){
-                console.log("BnetStrategy (2):", err);
-                return done(null, null);
-              });
+            log.debug("Updating existing user. 'profile':\n", profile);
           })
           .error(function(err){
-            console.log("BnetStrategy (1):", err);
-            return done(null, null);
+            log.debug("Creating new user. 'profile':\n", profile);
+          })
+          .finally(function(){
+            user.setValues({
+                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
+                oauthType: "bnet",
+                userId: profile.id
```

Comment:
Is there a reason why we are not just using `id` as the primary key?  It should be obvious that the `id` is for the user id.  Makes no real difference, but I want to join a UserProfile model to User, and it may be less confusing e.g. https://thinky.io/documentation/relations/